ota-plus-web/app/reactapp-new/src/components/common/Select.tsx

```diff
@@ -0,0 +1,116 @@
+import React, { ReactElement, useState } from 'react';
+import { Menu, Dropdown } from 'antd';
+import { Checkbox } from './';
+import styled, { DefaultTheme } from 'styled-components';
+import { useTranslation, UseTranslationResponse } from 'react-i18next';
+
+interface SelectOptions {
+  value: string;
+  row: () => ReactElement;
+}
+type Props = {
+  className?: string
+  options: SelectOptions[],
+  selected?: string[],
+  multiple?: boolean,
+  onSelect?: (p: string[]) => void,
+  dropdownWidth?: number | string,
+  width?: number | string,
+  placeholder?: string
+};
+type StyledProps = {
+  dropdownwidth?: number | string,
+  width?: number | string;
+  theme: DefaultTheme;
+};
+
+const StyledMenu = styled(Menu)<StyledProps>`
+  border-radius: 0;
+  box-shadow: ${({ theme }) => `${theme.shadows.light}`};
+  margin-top: -3px;
+  font-size: 16px;
+  width: ${({ dropdownwidth }) => `${dropdownwidth}`};
+  padding: 0;
+`;
+const StyledMenuItem = styled(Menu.Item)<StyledProps>`
+  border-bottom: ${({ theme }) => `1px solid ${theme.palette.accents.dark}`};
+  padding: 9px 10px;
+  &.ant-dropdown-menu-item-active {
+    background-color: ${({ theme }) => `${theme.palette.white}`};
+  }
+`;
+
+const StyledSelection = styled.div<StyledProps>`
+  background-color: ${({ theme }) => `${theme.palette.white}`};
+  box-shadow: ${({ theme }) => `${theme.shadows.light}`};
+  cursor: pointer;
+  display: inline-block;
```

Comment:
"inline-block is ignored due to the float. If 'float' has a value other than 'none', the box is floated and 'display' is treated as 'block'" - ts-styled-plugin